api/controllers/Depmu_EntryController.js

```diff
@@ -0,0 +1,93 @@
+/* global Prebooking DepmuEntryPrebookingValidatorService Movement */
+'use strict';
+
+var ValidationError = require('../lib/exceptions/ValidationError');
+var moment = require('moment-timezone');
+
+module.exports = {
+  _config: {
+    actions: false,
+    shortcuts: false,
+    rest: false,
+    exposedMethods: [
+      'prebooking'
+    ]
+  },
+
+  prebookingOptions: (req, res) => res.ok(DepmuEntryPrebookingValidatorService.schema),
+
+  truncatePrebookings: () =>
+    Prebooking.destroy({}),
+
+  formatPrebooking: prebooking => {
+    prebooking.cid_id = prebooking.cid_id ? parseInt(prebooking.cid_id) : null;
+    prebooking.location = prebooking.location.toLowerCase().trim();
+    prebooking.task_force = prebooking.task_force.toLowerCase().trim();
+    prebooking.timestamp = new Date(prebooking.timestamp);
+    return prebooking;
+  },
+
+  populatePrebookingWithCentreAndGender: prebooking =>
+    _.memoize(Centres.getGenderAndCentreByCIDLocation)(prebooking.location)
+      .then(result =>
+        _.merge(prebooking, result)),
+
+  filterNonEmptyPrebookings: prebooking =>
+  prebooking.centre && prebooking.gender && prebooking.task_force && prebooking.timestamp,
+
+  filterTodaysPrebookings: prebooking => {
+    var startOfDay = moment().set({hour: 7, minute: 0, second: 0, millisecond: 0});
```

Comment:
1. Even though I am aware that the Macro *currently* adds the same timestamp to each item in the payload; the reality is that each item has its own timestamp, meaning I would prefer for the API to not assume that each timestamp is the same ... basically for the API and Macro to be less coupled.

2. Yes, I've updated the name ... filterCurrentRangePrebookings